packages/jest-cli/src/__tests__/TestRunner-test.js

```diff
@@ -21,3 +37,47 @@ test('.addReporter() .removeReporter()', () => {
   runner.removeReporter(SummaryReporter);
   expect(runner._dispatcher._reporters).not.toContain(reporter);
 });
+
+
+describe('_createInBandTestRun()', () => {
+  test('injects the rawModuleMap to each the worker in watch mode', () => {
+    const config = {watch: true};
+    const rawModuleMap = jest.fn();
+    const hasteContext = {moduleMap: {getRawModuleMap: () => rawModuleMap}};
+
+    const runner = new TestRunner(hasteContext, config, {maxWorkers: 2});
+
+    return runner._createParallelTestRun(
```

Comment:
I decided to tests `_createParallelTestRun` instead of `runTests` even though it is a private method.

I did this because the current implementation of `runTests` is a harder to test, mainly because of all of the extra logic that the inner callbacks have. If you prefer I can also spend some time refactoring some of it to make `runTests` easier to test :)